src/core/FxMixer.cpp

```diff
@@ -197,6 +197,8 @@ FxMixer::FxMixer() :
 	// create master channel
 	createChannel();
 	m_lastSoloed = -1;
+
+	connect(this, SIGNAL(deletedChannel(int)), this, SLOT(deleteChannel(int)), Qt::QueuedConnection);
```

Comment:
`Qt::QueuedConnection` makes sure that when this signal is received the slot is activated only after the current event loop.

Thanks @curlymorphic for the hint.